File: output/nemotron-vs-qwen35-ud/20260326-raw/executor-agentic/workspaces/agentic_ts_restore_bootstrap_patch/qwen35_a3b_ud_q4_xl/repeat-1/apps/game-runtime/src/runtime/restore.ts
import type { RuntimeState } from "./types";
import { ThreadBootstrapper } from "./bootstrap";

export type Snapshot = {
  agentIds: string[];
  threadIds: Record<string, string>;
};

export function restoreRuntime(
  snapshot: Snapshot,
  state: RuntimeState,
  bootstrapper: ThreadBootstrapper,
): RuntimeState {
  state.threadIds = {};
  for (const agentId of snapshot.agentIds) {
    state.threadIds[agentId] = bootstrapper.ensureThread(agentId);
  }
  state.phase = "running";
  return state;
}

File: output/omnicoder-vs-27b/20260325-omnicoder/agentic/workspaces/agentic_python_restore_bootstrap/27b_ud_q4_xl/repeat-1/runtime/restore.py
from __future__ import annotations

from .session import Snapshot
from .threads import ThreadBootstrapper
from .types import RuntimeState


def restore(snapshot: Snapshot, state: RuntimeState, bootstrapper: ThreadBootstrapper) -> RuntimeState:
    state.phase = "running"
    state.thread_ids = {}
    for agent_id in snapshot.agent_ids:
        if agent_id not in state.thread_ids:
            state.thread_ids[agent_id] = bootstrapper.ensure_thread(agent_id)
    state.connected = True
    return state
